Package name not written in Java source files causes compilation error.

error: cannot find symbol
        public static PuzzleSolution solve(String initialConf, String goalConf, char distType) {
                      ^
  symbol:   class PuzzleSolution
  location: class Puzzle
 
and many more...

--- a/android/app/src/main/java/com/roadiq/Node.java
+++ b/android/app/src/main/java/com/roadiq/Node.java
@@ -1,3 +1,5 @@
+package com.roadiq;
+
 public class Node implements Comparable<Node> {
 	Node parent;
 	int state[][];

--- a/android/app/src/main/java/com/roadiq/Puzzle.java
+++ b/android/app/src/main/java/com/roadiq/Puzzle.java
@@ -1,3 +1,5 @@
+package com.roadiq;
+
 import java.util.*;
 
 public class Puzzle {
@@ -12,6 +14,8 @@ public class Puzzle {
 		String temp[] = null;
 
 		temp = initialConf.split(",");
+		if (temp.length != 9)
+			return new PuzzleSolution(-2);
 
 		int N = (int) Math.sqrt(temp.length);
 
@@ -28,6 +32,9 @@ public class Puzzle {
 				}
 
 		temp = goalConf.split(",");
+		if (temp.length != 9)
+			return new PuzzleSolution(-3);
+
 		int goal[][] = new int[N][N];
 		for (int i = 0, k = 0; i < N; i++)
 			for (int j = 0; j < N; j++, k++)

